src/resources.js

```diff
@@ -428,6 +463,7 @@ export class Resources {
         ', prerenderSize=', this.prerenderSize_);
     this.pass_.cancel();
     this.vsyncScheduled_ = false;
+    this.focusHistory_.purgeBefore(now - FOCUS_HISTORY_TIMEOUT_);
```

Comment:
This needs a bit of docs, but we might be able to simplify:

I believe with iframes the active element will be the iframe, so no descendant logic needed. In that case you could just record the id.